src/sfuncs/dist/cat.jl

```diff
@@ -1,7 +1,33 @@
-export Cat
+export 
+    Cat,
+    Categorical,
+    Discrete
 
 import ..Utils.normalize
-import Distributions.Categorical
+import Distributions
+
+const Categorical{P, Ps} = DistributionsSF{Distributions.Categorical{P, Ps}, Int}
+Categorical(p) = Categorical{typeof(p[1]), typeof(p)}(p)
+
+const Discrete{T,P,Ts,Ps} = DistributionsSF{Distributions.DiscreteNonParametric{T,P,Ts,Ps}, T}
+function Discrete(xs, ps)
+    # Handle duplicates
+    sort_order = sortperm(xs)
+    xs = xs[sort_order]
+    ps = ps[sort_order]
+
+    for i=1:(length(xs) - 1)
+        if xs[i] == xs[i + 1]
+            ps[i] += ps[i + 1]
+            ps[i + 1] = 0
+        end
+    end
+    keep = ps .> 0
+    xs = xs[keep]
+    ps = ps[keep]
+      
+    return Discrete{typeof(xs[1]), typeof(ps[1]), typeof(xs), typeof(ps)}(xs, ps)
```

Comment:
Assuming non-empty arrays/tuples is bad.

Reply:
Commented above.